### PR TITLE
Introduce MatCap shading to improve visual quality

### PR DESCRIPTION
**Pull request by christjt**

### CODE REVIEW

File: viewer/src/views/threejs/cad/matCapTextureData.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
...
...oLgP8Nvn+k/t8o/GPQWqMpKxGA23cH195S0qgekUdJkiT5bJekPO5RQMwTrM0qZS0kgGadYs2P13cle5lNwzSp2q0Mc5kdLmjb3AADPeVobOPaH6GJkgwT2gW0arbBIMhs0SPrCBjJi8k0VIyXxBxnIMO4lqJk4tzZmPBZrYBrAuVJl45QMUTb9U3OyqOm+fSbamweBNfitupfweH4sbobiMovy/Uf2+czxbuyrilPY/wAO4YlZNlxNlrnLO5yzH3nXi6oZ+TuyrSgrODMc8HiyqNIo6yUqKhADGNZ+mEX3ktUGg1bdjFX7Edbf5S24YA2XuZHRpAXG/SBQG1fSAgJqMT2NbYMpyx8LbA26VLAQ6g5lL7DgvTotXo9tHYDV18GwZT6d1+kG0xIo6TilNLBNWraVif8Aqbofk3/OJLTom1kUb7mwGPwn4WG4PyMaFIjWnbnbAM0xIa+x+zyp1kfI4V4maXk4sUNWPkRrJNhDtTE5g3BQMMYk+ZqkoBssCPnMWWShSAX6nPQyKU1UT9XZzDcx4Zei3gIiwnmI2VRuTsB8zIb2VljpEjV8Vp5yumRtU/TFeyD5t/xmPHXSKsRG3Q6vXjm1dmVznwkGE+vc/Wb4tPY8n5Ie0WjWkDlXH0lZbRg9FfT1+0PjymmN7Q2dPzD3nTVkoSmaqQjYzna8WVaG8PHSOUIaCA9JLRSfpmwpilGz7EIYJQy7SWAInAjg4Y5cxwGZeskbSkheUBNXviS9lIz4OTKaiB7GaKB6TJoGzer0dbpysoIxvtGmStkYcNt0jMeHamzTgndFOUPzU7ftLt6Ozg5pLOKVNmzS02+prJrJ+m4muOPtCTT6ULON6Ep4eqXU6Wwf9youv3XP8TH5mnzpf8nNPr9DYQKuJaV/bxQp+xxM1ro1ih3wdRapelRYvqjBh+0WL/qD8A+n0mtYH/AfHyl5vRPhsLbptRSoa8LWp7uwX+ZjUar43CPrOIaClsW8R0wx2FgY/YZmbbfC/BL2IXcX0p/yKtTfnpyVco+7Yj2zVrGaFLtVxLUginTU6dfVybG/2H8wX8vYN+S0Ivwy3UMG119l+Nwrnyj5KNpeWP0Qn5dHa9FXygBQMe0SRi/50bSkKeXAmmIJnfDCnIE2SpGQ1Ry9IkoyaPouRNsRM6UHXEMl5IaOqe0zThSZsLvLexhVq5u28iDTP//Z'
```
Comment (larsmoa) [inline viewer/src/views/threejs/cad/matCapTextureData.ts:3]: Add a reference to https://github.com/nidorx/matcaps as source of the image (https://github.com/nidorx/matcaps/blob/master/PAGE-26.md)

File: viewer/src/views/threejs/cad/materials.ts [hunk 2]
```diff
@@ -39,6 +40,9 @@ export function createMaterials(treeIndexCount: number): Materials {
   visibility.fill(255);
   const overrideColorPerTreeIndex = new THREE.DataTexture(colors, textureDims.width, textureDims.height);
   const overrideVisibilityPerTreeIndex = new THREE.DataTexture(visibility, textureDims.width, textureDims.height);
+  
+  var matCapTexture = new THREE.Texture(matCapTextureImage);
```
Comment (larsmoa) [inline viewer/src/views/threejs/cad/materials.ts:44]: var -> const

File: viewer/src/views/threejs/cad/materials.ts [hunk 3]
```diff
@@ -39,6 +40,9 @@ export function createMaterials(treeIndexCount: number): Materials {
...
+  matCapTexture.needsUpdate = true;
```
Comment (larsmoa) [inline viewer/src/views/threejs/cad/materials.ts:45]: is setting needsUpdate necessary for new textures?

File: viewer/src/views/threejs/cad/matCapTextureData.ts (hunk 1 above)
Comment (larsmoa) [reply viewer/src/views/threejs/cad/matCapTextureData.ts:3]: Also, size is 232x232 and ThreeJS reduces size to 128x128 runtime. Reduce size of the encoded image to 128x128.

**Comment by dragly** [comment]
The pull request should probably have a different title to make sure it does not get merged with the current one as commit message.

File: viewer/src/views/threejs/cad/materials.ts (hunk 3 above)
Comment (christjt) [reply viewer/src/views/threejs/cad/materials.ts:45]: Well, everything just turns black if I don't have this. So I assume you need it.

**Pull request merged**